core/lib/Thelia/Core/Stack/ParamInitMiddleware.php

```diff
@@ -134,7 +134,14 @@ protected function defineCurrency(TheliaRequest $request)
     protected function detectLang(TheliaRequest $request)
     {
         // first priority => lang parameter present in request (get or post)
-        if ($request->query->has("lang")) {
+        $requestedLangCodeOrLocale = $request->query->has("lang");
+
+        // add a fallback on locale parameter
+        if (null === $requestedLangCodeOrLocale) {
+            $requestedLangCodeOrLocale = $request->query->has("locale");
+        }
+
+        if (null !== $requestedLangCodeOrLocale) {
             // The lang parameter may contains a lang code (fr, en, ru) for Thelia < 2.2,
             // or a locale (fr_FR, en_US, etc.) for Thelia > 2.2.beta1
             $requestedLangCodeOrLocale = $request->query->get("lang");
```

Comment:
You have to do something here, this line is wrong and depends on what you will do with the lines above.